projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+)
+
+func sendRequest(url string) *http.Response {
+	resp, err := http.Get(url)
+	if err != nil {
+		fmt.Println("We can't get the weather!")
+		fmt.Fprintln(os.Stderr, "Failed to initialize.!", err)
+		os.Exit(2)
+	}
+	if resp == nil {
+		fmt.Fprintln(os.Stderr, "The server did not reply anything, which here is considered an error!")
+		os.Exit(52)
+	}
+	return resp
+}
+
+func handleStatusCode(resp *http.Response, sleepTime *time.Duration) {
```

Comment:
Why does this accept a pointer to a `time.Duration` as an argument? I'd definitely add some documentation to this function explaining what value is expected to be passed there, and how it may be changed by the function.

As far as I can tell, you're using this is a way for you to return a value from the function? In which case, I'd just return the value, rather than populate a pointer?

But also, the caller doesn't use it, as the sleeping is done inside this function, so I'm not sure why it's returning a value to the caller at all?

Reply:
I needed to keep total of sleepTimes to prevent sleeping and retrying for long time. I will change it to return a value.
I think you suggest return rather than pointer because of speed?